src/testshade/testshade.cpp

```diff
@@ -2254,7 +2254,7 @@ test_shade(int argc, const char* argv[])
             std::cout << texturesys->getstats(5) << "\n";
         std::cout << ustring::getstats() << "\n";
     }
-    if (debug1 || print_groupdata) {
+    if ((debug1 || print_groupdata) && !batched) {
         int groupdata_size;
```

Comment:
Does this mean that llvm_groupdata_size was not being set properly on the batched side?  Is that fixable (although might be outside the scope of your PR?.  If so can we leave a TODO comment in the code to fix it?

Reply:
Step one I just wanted to pass these CI tests :)

I can remove that if-clause, and then when batched == true I can output llvm_groupdata_wide_size. But am I correct in thinking that number will vary with architecture? I will need a ref/out.txt for each possible output if so.

Reply:
@chellmuth correct on both counts.  Typically 8wide and 16wide will have the same sizes/outputs.  But perhaps for AVX512 some integers only used as booleans might only take 2 bytes because they use native bit masks.

Reply:
I think I'll follow whatever you recommend then. I can enable the groupdata size output, run the CI tests, and add new ref.txt's from the different architectures' unique failures. Or I can omit the line. Happy either way :)

Reply:
To make life easier maybe just the TODO, but could the comparison can just verify the "optimized" group data is smaller than non-opt vs. specific sizes.

Reply:
Sounds good. Unfortunately with how the testsuite is setup, I don't know any precedent for that kind of assert.

Reply:
ok, lets just do the TODO and move on.  
Thanks for your work on this.